llvm/lib/CodeGen/MachineStableHash.cpp

```diff
@@ -97,9 +99,14 @@ stable_hash llvm::stableHashValue(const MachineOperand &MO) {
       ++StableHashBailingGlobalAddress;
       return 0;
     }
-    auto Name = GV->getName();
-    return stable_hash_combine(MO.getType(), MO.getTargetFlags(),
-                               stable_hash_name(Name), MO.getOffset());
+    stable_hash GVHash = 0;
+    if (auto *GVar = dyn_cast<GlobalVariable>(GV))
```

Comment:
Seems like we can generate meaningful hash for globals regardless of the presence of name. Should we move the `hasName` check after this?

Reply:
Moved `hasName` check below.